src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -265,6 +267,352 @@ protected override DbCommand CreateDbCommand()
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+            if (State != ConnectionState.Open)
+            {
+                throw new InvalidOperationException(Resources.CallRequiresOpenConnection(callerName));
+            }
+
+            var rc = raw.sqlite3_create_function(_db, name, numberOfArguments, state, scalarFunction);
+            SqliteException.ThrowExceptionForRC(rc, _db);
+        }
+
+        /// <summary>
+        /// Create user defined aggregate.
+        /// </summary>
+        /// <typeparam name="TContext">Type of the state buffer.</typeparam>
+        /// <param name="name">Name of the aggregate.</param>
+        /// <param name="function">The user defined aggregate.</param>
+        /// <returns>The seed value for the state buffer.</returns>
+        public virtual TContext CreateAggregate<TContext>(
+            string name,
+            Func<TContext, TContext> function)
+            => CreateAggregate(name, default(TContext), function);
+
+        /// <summary>
+        /// Create user defined aggregate.
+        /// </summary>
+        /// <typeparam name="TContext">Type of the state buffer.</typeparam>
+        /// <param name="name">Name of the aggregate.</param>
+        /// <param name="seed">Seed for initializing the aggregate.</param>
+        /// <param name="function">The user defined aggregate.</param>
+        /// <returns>The seed value for the state buffer.</returns>
+        public virtual TContext CreateAggregate<TContext>(
+            string name,
+            TContext seed,
+            Func<TContext, TContext> function)
+            => CreateAggregate(name, seed, function, (state) => state);
+
+        /// <summary>
+        /// Create user defined aggregate.
+        /// </summary>
+        /// <typeparam name="TContext">Type of the state buffer.</typeparam>
+        /// <typeparam name="TResult">Type returned by the result selector.</typeparam>
+        /// <param name="name">Name of the aggregate.</param>
+        /// <param name="seed">Seed for initializing the aggregate.</param>
+        /// <param name="function">The user defined aggregate.</param>
+        /// <param name="resultSelector">The user defined result selector.</param>
+        /// <returns>The result based on the seed value for the state buffer.</returns>
+        public virtual TResult CreateAggregate<TContext, TResult>(
+            string name,
+            TContext seed,
+            Func<TContext, TContext> function,
+            Func<TContext, TResult> resultSelector)
+        {
+            delegate_function_aggregate_step stepFunction = null;
+            delegate_function_aggregate_final finalFunction = null;
+            if (function != null)
+            {
+                stepFunction = (ctx, _, args) =>
+                {
+                    if (ctx.state == null)
+                    {
+                        ctx.state = seed;
```

Comment:
If the result can be `null`, this may cause problems. One way around it would be to wrap the value in an type so we know if we've initialized it or not.
```C#
if (ctx.state == null)
    ctx.state = new MyHolder<TContext> { Value = seed };

ctx.state.Value = function(ctx.state.Value);
```